rasa/shared/core/domain.py

```diff
@@ -1683,10 +1725,39 @@ def _validate_slot_mappings(forms: Union[Dict, List]) -> None:
     if not isinstance(forms, dict):
         raise InvalidDomain("Forms have to be specified as dictionary.")
 
-    for form_name, slots in forms.items():
-        if slots is None:
+    for form_name, form_data in forms.items():
+        if form_data is None:
             continue
 
+        if isinstance(form_data, Dict):
+            if "required_slots" in form_data:
+                slots = forms[form_name].get("required_slots")
+            elif "global_not_intent" in form_data and "required_slots" not in form_data:
+                raise InvalidDomain(
+                    f"If you use the `global_not_intent` parameter in your form, then"
+                    f" the keyword `required_slots` should precede the definition"
+                    f" of your slot mappings. Please see"
+                    f" {rasa.shared.constants.DOCS_URL_FORMS} for more information."
+                )
+            else:
+                slots = form_data
+        else:
```

Comment:
I think we still need to treat the `list` type as valid: https://github.com/RasaHQ/rasa/blob/b9705f455d05cc619c222d16970f2b412fae3af1/rasa/shared/core/domain.py#L635

Reply:
So I will remove the warning but if the form is a list does this affect how I retrieve the `slots` or how I check the other parameters (`global_not_intent` and `required_slots`) ?

Reply:
You can safely remove it 👍🏻 Any forms defined as lists will have no mapped slots anyway. `_initialize_forms` takes care of this